lib/map-file.js

```diff
@@ -435,11 +435,14 @@ class MapFileReader {
      * @param {array of objects} spaces
      * @param {object} address 
      */
-    isWithinAddressSpace(spaces, address) {
+    isWithinAddressSpace(spaces, address, size) {
         for (let idxSpace = 0; idxSpace < spaces.length; ++idxSpace) {
             const spaceAddress = spaces[idxSpace];
 
-            if ((address >= spaceAddress.startAddress) && (address < spaceAddress.endAddress)) {
+            if (((address >= spaceAddress.startAddress) &&
+                 (address < spaceAddress.endAddress)) || 
```

Comment:
This check took me a while to double- and triple- check :) But...it is right =) Nice job!

Reply:
should probably have a unittest for it just in case..